pkg/querier/tripperware/queryrange/query_range.go

```diff
@@ -159,8 +182,14 @@ func (prometheusCodec) EncodeRequest(ctx context.Context, r tripperware.Request)
 		}
 	}
 
-	// Always ask gzip to the querier
-	h.Set("Accept-Encoding", "gzip")
+	if c.compression == GzipCompression {
+		h.Set("Accept-Encoding", string(c.compression))
+	}
+	if c.enableProtobuf {
+		h.Set("Accept", applicationProtobuf)
+	} else {
+		h.Set("Accept", applicationJson)
+	}
```

Comment:
Can we put this logic to a common package (for example, tripperware) so that it can be shared by both query range and instant query?
Same for the logic used in decoding response

Reply:
I moved it to `query.go`. It looks much cleaner now but let me know if you prefer moving it to another file in tripperware